### PR TITLE
[knife] add order by and limit the number of cookbooks returned when listing

### PR DESCRIPTION
**Pull request by iennae**
### Description

This updates knife cookbook site list to add order by and limit number functionality. It changes the default behavior from listing all cookbooks to 100. The refactor speeds up the response.

Additionally, code is removed from knife cookbook show list that isn't used. 

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] RELEASE\_NOTES.md, has been updated if required (not required for bugfixes, required for API changes)
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by iennae** [comment]
@robbkidd

**Comment by thommay** [comment]
@iennae please fix the chefstyle violations in travis

File: lib/chef/knife/cookbook_site_show.rb [hunk 1]
```diff
@@ -47,20 +47,6 @@ def get_cookbook_data
...
-        cr["items"].each do |cookbook|
-          cookbook_collection[cookbook["cookbook_name"]] = cookbook
-        end
-        new_start = start + cr["items"].length
-        if new_start < cr["total"]
-          get_cookbook_list(items, new_start, cookbook_collection)
-        else
-          cookbook_collection
-        end
-      end
```
Comment (robbkidd) [inline lib/chef/knife/cookbook_site_show.rb]: tenor-151493477

**Comment by robbkidd** [comment]
👍  These were the changes @sigje and I paired on.

**NOTE:** default behavior changed. With this change, `knife cookbook site list` won't list _all_ cookbooks at the given Supermarket URL. Defaults to listing the first 100 of a given sort order. If changing the default behavior isn't acceptable, something could be done with some new option switches and two API trips: one to get the total and another to retrieve that number items.

**Comment by stevendanna** [comment]
Hard to say whether this behavior change is the right one without know how people use that command.  Some thoughts:

- By removing the pagination, it means that if someone sets `-n 5000` to ensure they can list all of the cookbooks on the community site, then we make a single request asking for 5000 items.  Do we think that would be problematic for public or private supermarket sites?

- An alternative might be to keep the pagination (increasing the page size to 100 or more) and then also adding a configurable overall limit.

**Comment by thommay** [comment]
I don't think we can change default behaviour in a minor version.

**Comment by robbkidd** [comment]
> By removing the pagination, it means that if someone sets -n 5000 to ensure they can list all of the cookbooks on the community site, then we make a single request asking for 5000 items. Do we think that would be problematic for public or private supermarket sites?

The Supermarket v1 API currently allows someone to make a request with items = 5000. At the current number of cookbooks on public Supermarket, this hasn't been a problem.

> An alternative might be to keep the pagination (increasing the page size to 100 or more) and then also adding a configurable overall limit.

Another alternative is experimenting with chunked transfer encoding. I don't know offhand how the rest client inside chef/knife handles that, though.

**Pull request closed without merging**